File: Array/2996. Smallest Missing Integer Greater Than Sequential Prefix Sum/missingInteger.go
package main

import "fmt"

func missingInteger(nums []int) int {
	sum := nums[0]
	var prefix bool
	for i := 1; i < len(nums); i++ {
		if nums[i-1] == nums[i]-1 {
			sum += nums[i]
		} else {
			break
		}
	}
	for {
		prefix = true
		for i := 0; i < len(nums); i++ {
			if nums[i] == sum {
				sum++
				prefix = false
			}
		}
		if prefix {
			break
		}
	}
	return sum
}

// 示例用法
func main() {
	nums := []int{1, 2, 3, 5}
	result := missingInteger(nums)
	fmt.Println("Missing integer:", result)
}

File: Array/2996. Smallest Missing Integer Greater Than Sequential Prefix Sum/missingInteger.c
#include <stdio.h>

int missingInteger(int* nums, int numsSize) {
    int sum = nums[0];
    int prefix;
    for (int i = 1; i < numsSize; i++) {
        if (nums[i - 1] == nums[i] - 1) {
            sum += nums[i];
        } else {
            break;
        }
    }
    while (1) {
        prefix = 1;
        for (int i = 0; i < numsSize; i++) {
            if (nums[i] == sum) {
                sum++;
                prefix = 0;
            }
        }
        if (prefix == 1) {
            break;
        }
    }
    return sum;
}

// 示例用法
int main() {
    int nums[] = {1, 2, 3, 5};
    int numsSize = sizeof(nums) / sizeof(nums[0]);
    int result = missingInteger(nums, numsSize);
    printf("Missing integer: %d\n", result);
    return 0;
}